lib/kubernetes-deploy/kubernetes_resource.rb

```diff
@@ -38,7 +38,11 @@ def build(namespace:, context:, definition:, logger:, statsd_tags:, crd: nil)
         opts = { namespace: namespace, context: context, definition: definition, logger: logger,
                  statsd_tags: statsd_tags }
         if (klass = class_for_kind(definition["kind"]))
-          return klass.new(**opts)
+          if klass <= CustomResource
```

Comment:
I think you could remove this, and burn both Memcached and Redis instead no? I guess this is for compatibility reason only?

Reply:
Memcached and Redis check also the deployment (nb replicas == available pods).  We need to keep this because `CustomResource` does not handle that.

```ruby
   def deploy_succeeded?
      return false unless super
      deployment_ready? && service_ready?
    end
```

Reply:
Would it be possible to make the buddy check these conditions. We really want to 🔥 these files since they are in an open source repo, but are only useable to shopify.

Reply:
> Would it be possible to make the buddy check these conditions?
Let's go for that.